portable/ThirdParty/GCC/Posix/port.c

```diff
@@ -119,6 +121,44 @@ static void vPortStartFirstTask( void );
 static void prvPortYieldFromISR( void );
 /*-----------------------------------------------------------*/
 
+void prvThreadKeyDestructor( void * data )
+{
+    free( data );
+}
+
+static void prvInitThreadKey()
+{
+    pthread_mutex_lock( &xThreadMutex );
+
+    if( xThreadKey == 0 )
+    {
+        pthread_key_create( &xThreadKey, prvThreadKeyDestructor );
+    }
+
+    pthread_mutex_unlock( &xThreadMutex );
+}
+
+static void prvMarkAsFreeRTOSThread( pthread_t thread )
+{
+    prvInitThreadKey();
+    uint8_t * thread_data = malloc( 1 );
```

Comment:
could you add configASSERT( thread_data != NULL) after the malloc call? I imagine it'd be fairly unlikely that malloc would return NULL when you are asking for 1 byte, but if it did, the FreeRTOS thread would be marked as a non-FreeRTOS thread based on your logic and you'd experience some pretty weird behavior I imagine.

Reply:
Sure

Reply:
Done

Reply:
Correction to my previous statement - it would still crash when you dereferenced on the next line - the assert at least halts the program and makes it clear why the program crashed though so thank you.